spec.bs

```diff
@@ -1244,26 +1328,32 @@ an [=ordered map=] whose [=map/keys=] are [=strings=] and whose [=map/values=] a
 (initially null).
 
 <div algorithm>
-  To <dfn>evaluate a bidding script</dfn> given a [=string=] |script|, an [=interest group=] |ig|,
-  a [=list=] of arguments |arguments|, and an integer millisecond duration |timeout|:
+  To <dfn>evaluate a bidding script</dfn> given a [=string=] |script|, an [=interest group=] |ig|, a
+  {{GenerateBidInterestGroup}} |igGenerateBid|, a [=string=]-or-null |auctionSignals|, an
+  [=ordered map=]-or-null |perBuyerSignals|, an [=ordered map=] |trustedBiddingSignals|, an
+  [=ordered map=] |browserSignals|, and an integer millisecond [=duration=] |timeout|:
 
-    1. Let |groupHasAdComponents| be true.
-    1. If |ig|'s [=interest group/ad components=] is null:
-      1. Set |groupHasAdComponents| be false.
     1. Let |global| be a new {{InterestGroupBiddingScriptRunnerGlobalScope}}.
     1. Set |global|'s
-      [=InterestGroupBiddingScriptRunnerGlobalScope/group has ad components=] to
-      |groupHasAdComponents|.
-    1. Let |numArguments| be |arguments|'s [=list/size=].
-    1. Let |browserSignals| be |arguments|[|numArguments| - 1].
-    1. Let |isComponentAuction| be false.
-    1. If |browserSignals|["`topLevelSeller`"] is not null:
-      1. Set |isComponentAuction| to true.
+      [=InterestGroupBiddingScriptRunnerGlobalScope/group has ad components=] to true if |ig|'s
+      [=interest group/ad components=] is not null, or false otherwise.
     1. Set |global|'s
-      [=InterestGroupBiddingScriptRunnerGlobalScope/is component auction=] to |isComponentAuction|.
+      [=InterestGroupBiddingScriptRunnerGlobalScope/is component auction=] to true if
+      |browserSignals|["`topLevelSeller`"] is not null, or false otherwise.
     1. Set |global|'s [=InterestGroupBiddingScriptRunnerGlobalScope/interest group=] to |ig|.
+    1. Let |igJS| be the result of [=converted to ECMAScript values|converting to ECMAScript values=]
+      given |igGenerateBid|.
+    1. Let |auctionSignalsJS| be the result of [=converted to ECMAScript values|converting to ECMAScript values=]
+      given |auctionSignals|.
+    1. Let |perBuyerSignalsJS| be the result of [=converted to ECMAScript values|converting to ECMAScript values=]
+      given |perBuyerSignals|.
+    1. Let |trustedBiddingSignalsJS| be the result of [=converted to ECMAScript values|converting to ECMAScript values=]
+      given |trustedBiddingSignals|.
+    1. Let |browserSignalsJS| be the result of [=converted to ECMAScript values|converting to ECMAScript values=]
+      given |browserSignals|.      
     1. Return the result of [=evaluating a script=] with |global|, |script|, "`generateBid`",
-      |arguments|, and |timeout|.
+      [ |igJS|, |auctionSignalsJS|, |perBuyerSignalsJS|, |trustedBiddingSignalsJS|, |browserSignalsJS| ],

```

Comment:
@jyasskin This looks weird to me. Can I construct an EMCA Array object like this and pass it to [=evaluate a script=] method?
Or create a [=list=] of the JS arguments and then convert the [=list=] to an EMCA Array object in [=evaluate a script=] method, which also sounds weird.

Reply:
WebIDL uses the https://webidl.spec.whatwg.org/#web-idl-arguments-list type to bundle up IDL-typed things, and then https://webidl.spec.whatwg.org/#web-idl-arguments-list-converting to create the ES argument list.

[Infra](https://infra.spec.whatwg.org/#lists) and [ES Lists](https://tc39.es/ecma262/#sec-list-and-record-specification-type) are the same thing, denoted with « ... », but both are a different sort of thing from an ES Array object.